mailgreen/controller/mail_controller.py

```
@@ -163,3 +182,96 @@ async def get_sender_details(
+            errors.append(
+                {
+                    "msg_id": gmail_msg_id,
+                    "error": "DB에서 해당 Gmail 메시지 ID를 찾을 수 없습니다.",
+                }
+            )
+
+    try:
+        db.commit()
+    except Exception as db_err:
+        db.rollback()
+        raise HTTPException(status_code=500, detail=f"DB 업데이트 실패: {str(db_err)}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)